src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -269,9 +271,23 @@ protected Object invokeStep(StepDescriptor d, String name, Object args) {
 
             // Persist the node - block start and end nodes do their own persistence.
             CpsFlowExecution.maybeAutoPersistNode(an);
-            StepExecution e = s.start(context);
-            thread.setStep(e);
-            sync = e.start();
+
+            // Call any registered StepListeners.
+            for (StepListener sl : ExtensionList.lookup(StepListener.class)) {
+                try {
+                    sl.notifyOfNewStep(s, context);
+                } catch (Exception e) {
```

Comment:
Seems like a good idea to be extra careful like with `GraphListener` methods and catch `Throwable`.

```suggestion
                } catch (Throwable e) {
```